### PR TITLE
Bringing in pvoutput publishing and minor improvements

### LINKED ISSUE #22: PVOutput support
Enhancement to upload Solariot metrics to pvoutput.org.

API specs: https://pvoutput.org/help.html#api-spec

### PR DESCRIPTION
**Pull request by michael-robbins**
This was pulled from https://github.com/corvax/solariot/commit/d4146205acde600233eca1d967c86178337bda18
So credits to @corvax for the hard work!
I've just tweaked it by wrapping it in a class to make it a bit more modular/portable as well as expanding on the other `vX` parameters you can provide to pvoutput!

Additional features as I was doing this all at once:
* Modbus maps can now have `metric_100` or `metric_10000` to have the script divide by that figure to get you to the target decimal (this can even be say... `metric_0.01` to move the decimal place to the right!)
* Updates to the sg5kd.py to track a few more metrics and adding comments to the 'units' these metrics produce (W vs kW vs MWh etc), I've made a number of these from `kW => W` because PVOutput tracks stuff in W
* Made the solariot.py script default to python3 now as we kinda need it
* Improved MQTT publishing by reconnecting before sending, as after a while I was seeing MQTT_ERR_NO_CONN which indicated the client dropped the connection to the broker
* Adding a .dockerignore file that will not accidentally include any potentially sensitive files into your locally built docker image ;)

This closes #22

### CODE REVIEW

File: solariot.py [hunk 1]
```diff
@@ -141,6 +141,55 @@
     flux_client = None
     logging.info("No InfluxDB configuration detected")
 
+# Configure PVOutput
+if hasattr(config, "pvoutput_api"):
+    class PVOutputPublisher(object):
+        def __init__(self, api_key, system_id, status_url="https://pvoutput.org/service/r2/addstatus.jsp"):
```
Comment (michael-robbins) [inline solariot.py:147]: We bring the URL in here as we've kinda tied the payload below to the 'r2' of the API anyway, so no point making that configurable in the config file as we'd have to probably change the internals of this as well!

**Comment by michael-robbins** [comment]
@meltaxa this is good to test & merge! It's working on my system and publishing to pvoutput (one data point so far, will see after a few hours how it's looking)

**Comment by michael-robbins** [comment]
Ok pushed up a few more random fixes to timeouts/disconnects that I encountered testing out this feature!

**Comment by meltaxa** [comment]
Looks good. I have made a change to allow each modbus file to also map to it's respective values for pvoutput. See my version of your branch feature/pvoutput-publishing for comparison.

**Comment by meltaxa** [comment]
Added PVOutput rate limit support.

**Pull request merged**